### PR TITLE
Pulling in commits from upstream and adding fix for Rails 6

### PR DESCRIPTION
**Pull request by zackreverb**
... to make this compatible with our code base

### CODE REVIEW

File: lib/attr_encrypted/adapters/active_record.rb [hunk 1]
```diff
@@ -76,7 +76,7 @@ def attr_encrypted(*attrs)
               # attributes are handled, @attributes[attr].value is nil which
               # breaks attribute_was. Setting it here returns us to the expected
               # behavior.
-              if Gem::Requirement.new('>= 5.2').satisfied_by?(RAILS_VERSION)
+              if RAILS_VERSION < Gem::Version.new(6.0) && RAILS_VERSION >= Gem::Version.new(5.2)
```
Comment (zackreverb) [inline lib/attr_encrypted/adapters/active_record.rb:79]: This is the key to compatibility with our repo.

**Comment by zackreverb** [comment]
Hey @wenn could you give this a look? This includes all of the commits to get us to the latest SHA of the upstream repo, plus the one line change to make this play nicely with Rails 6. I deployed this to unstable, viewed a TIN, then wrote a TIN and verified we got the same behavior as the current SHA. It looks good.

I tested on unstable with this change: https://github.com/reverbdotcom/reverb/pull/36041

File: lib/attr_encrypted/adapters/active_record.rb (hunk 1 above)
Comment (wenn) [reply lib/attr_encrypted/adapters/active_record.rb:79]: Do we need this block? We're on rails 6+ no?

Comment (zackreverb) [reply lib/attr_encrypted/adapters/active_record.rb:79]: That's true, we could pull it out entirely. The only important thing for us is that `          set_attribute_was(attr,__send__(attr))` is not called. This was just the fix suggested upstream that I copied over. Could delete the entire thing if there's a preference for that.

**Review by wenn — APPROVED** [review]

File: lib/attr_encrypted/adapters/active_record.rb (hunk 1 above)
Comment (wenn) [reply lib/attr_encrypted/adapters/active_record.rb:79]: Gotcha, no a big deal - we're diverging from the main line anyways.

**Pull request merged**